libs/datamodel/core/src/transform/ast_to_dml/db/walkers/index.rs

```diff
@@ -24,7 +24,16 @@ impl<'ast, 'db> IndexWalker<'ast, 'db> {
 
         let model = self.db.walk_model(self.model_id);
         let model_db_name = model.final_database_name();
-        let field_db_names: Vec<&str> = model.get_field_db_names(&self.index_attribute.fields).collect();
+        let field_db_names: Vec<&str> = model
+            .get_field_db_names(
+                &self
+                    .index_attribute
+                    .fields
+                    .iter()
+                    .map(|f| f.field_id)
+                    .collect::<Vec<_>>(),
```

Comment:
I don't understand yet why this iter + collect and the one below are necessary

Reply:
I think I understand now. It's an unfortunate side effect of the FieldWithArgs. The Vec<FieldId> was already less than ideal and I've had ideas for a better abstraction for a long time. Maybe soon.

Reply:
the methods expect &[FieldId] but the fields are now a Vec<FieldWithArgs> since they need to hold the sort and length arg in addition to the FieldId